packages/frontend-2/components/settings/Dialog.vue

```diff
@@ -141,80 +130,26 @@ const { result: workspaceResult } = useQuery(settingsSidebarWorkspacesQuery, nul
 const isMobile = breakpoints.smaller('md')
 const targetWorkspaceId = ref<string | null>(null)
 
-const menuItemConfig = shallowRef<{ [key: string]: { [key: string]: MenuItem } }>({
-  user: {
-    [settingsQueries.user.profile]: {
-      title: 'Profile',
-      component: SettingsUserProfile
-    },
-    [settingsQueries.user.notifications]: {
-      title: 'Notifications',
-      component: SettingsUserNotifications
-    },
-    [settingsQueries.user.developerSettings]: {
-      title: 'Developer settings',
-      component: SettingsUserDeveloper
-    }
-  },
-  server: {
-    [settingsQueries.server.general]: {
-      title: 'General',
-      component: SettingsServerGeneral
-    },
-    [settingsQueries.server.projects]: {
-      title: 'Projects',
-      component: SettingsServerProjects
-    },
-    [settingsQueries.server.activeUsers]: {
-      title: 'Active users',
-      component: SettingsServerActiveUsers
-    },
-    [settingsQueries.server.pendingInvitations]: {
-      title: 'Pending invitations',
-      component: SettingsServerPendingInvitations
-    }
-  },
-  workspace: {
-    members: {
-      title: 'Members',
-      component: SettingsWorkspacesMembers
-    }
-  }
-})
-
-const isOpen = defineModel<boolean>('open', { required: true })
-const targetMenuItem = defineModel<string | null>('targetMenuItem', { required: true })
-
-const workspaceItems = computed(() =>
-  workspaceResult.value?.activeUser
-    ? workspaceResult.value.activeUser.workspaces.items
-    : []
+const workspaceItems = computed(
+  () => workspaceResult.value?.activeUser?.workspaces.items ?? []
 )
 const hasWorkspaceItems = computed(() => workspaceItems.value.length > 0)
 const isAdmin = computed(() => user.value?.role === Roles.Server.Admin)
-const selectedMenuItem = computed((): MenuItem | null => {
+const selectedMenuItem = computed((): SettingsMenuItem | null => {
   const categories = [
-    menuItemConfig.value.user,
-    menuItemConfig.value.server,
-    menuItemConfig.value.workspace
+    userMenuItems.value,
+    serverMenuItems.value,
+    workspaceMenuItems.value
   ]
   for (const category of categories) {
     if (targetMenuItem.value && targetMenuItem.value in category) {
       return category[targetMenuItem.value]
     }
   }
 
-  if (!isMobile.value && targetMenuItem.value) {
```

Comment:
Removed this because it was more annoying to add up on a random page, and not having a `targetMenuItem` is possible behavior